WordPress/gutenberg
# Improve consistency of the Post editor and Site editor Document actions

This PR seeks to address part of https://github.com/WordPress/gutenberg/issues/51460 and https://github.com/WordPress/gutenberg/issues/51729 

## What?

The 'document title' / 'document actions' in the top bar are inconsistent between the Post editor and the Site editor, especially in 'editing template' mode.
- The post editor 'document title' renders a third button that does the same thing of the second button: they both open the command centre.
- In the Site editor, the buttons focus style is missing.
- The hover style is set on the entire thing: when hovering one button, also the second button gets a hover style. 
- The command centre button renders long text inconsistently in the Post and Site editor.
- The Site editor 'Back' button doesn't handle the chevron icon for RTL languages. Screenshot:

<img width="552" alt="Screenshot 2023-07-03 at 14 52 39" src="https://github.com/WordPress/gutenberg/assets/1682452/3a7267ab-a128-4f80-abcd-507a26cba767">


## Why?

- The document title / actions in the Site editor and post editor are very similar UIs and should look and behave as close as possible.
- A duplicate button that does the same thing of another adjacent button is pointless and just confusing for users.
- Any control should have a clear indication of keyboard focus.
- The hover style should indicate only one control at a time: the one that is actually hovered.
- Long text overflow should behave consistently.

## How?

- Removes a redundant button in the Post editor.
- Adds missing focus style in the Site editor.
- Makes the hover style highlight only the control that is actually hovered.
- Adjusts the CSS to make text overflow behave consistently.
- Renames the edit post `DocumentTitle` component to `DocumentActions` for consistency. Also, this component is not just the 'title'. It does more than that so the name `DocumentActions` seems more appropriate.
- Adjust the CSS to be way more consistent between the two components.
- Makes the Site editor 'Back' handle the chevron icon for RTL languages.

## About the CSS changes

Worth noting that on trunk the DocumentTitle / DocumentActions already have some layout issues, especially on small screens. 

On desktop, long text overflow behaves differently. Screenshot of Post editor vs. Site editor:

<img width="1184" alt="Screenshot 2023-07-03 at 11 46 16" src="https://github.com/WordPress/gutenberg/assets/1682452/bfda1f6c-8daa-4d83-b850-2440d432904d">

The text overflow should work way better with this PR.

This PR does _not_ aim to fix the layout breakage on small screens, especially in the Post editor. This should be addressed separately. Screenshot from trunk:

With long text:

<img width="1187" alt="Screenshot 2023-07-03 at 11 47 26" src="https://github.com/WordPress/gutenberg/assets/1682452/50a06397-f76d-4b9a-a703-c1b6e958f34a">

Post editor, shorter text on even smaller screen:

<img width="599" alt="Screenshot 2023-07-03 at 14 31 54" src="https://github.com/WordPress/gutenberg/assets/1682452/41862605-dd98-4c11-ad57-8d7545e9bf4c">

## Testing Instructions




- Use a block-baseed theme e.g. Twenty Twenty-Three
- Edit a post.
- In the Settings panel, click the template name. A popover appears. Click 'Edit template'.
- Open a new browser tab.
- Go to the Site editor > Design > Pages > Any page. In the settings sidebar, click 'Edit template'.
- Compare the two editors top bar. In this state, both the editor 'document actions' in the center of the top bar show a 'Back' button and the Command centre button.
- In the Post editor, observe `⌘K` is not a separate button any longer.
- In both editors, tab through the buttons and observe they have a focus style.
- In both editors, check the hover style:
  - Hover the Back button: observe only the Back button has a hover style.
  - Hover the Command centre button: observe only the Command centre button has a hover style.
- In the dev tools inspector, change the Command centre button visible text to a longer text. I used: `The quick, brown fox jumps over a lazy dog. DJs flock by when MTV ax quiz prog`
- Observe the text takes all the available space and it does show the CSS overflow ellipsis.
- Switch the WP admin to a RTL language. Observe in the Site Editor the 'Back' button now ues the chevron-right icon.

As said above, on small screens the layout isn't great and breaks. It already happens on trunk and should be addressed separately. Not in the scope of this PR.


### Testing Instructions for Keyboard


## Screenshots or screencast
